### PR TITLE
feat: Add additional power field types for IP address, LXD, and Virsh

### PR DESCRIPTION
**Pull request by ndv99**
## Done
- Added LXD, Virsh, and IP address to power field types (in line with new backend types)
- Added simple IP address validaiton (check if field value is a valid IP address)

## QA steps

- [ ] Go to a machine with IPMI power type (details page, "Configuration")
- [ ] Click "Edit" under "Power configuration"
- [ ] Ensure the IP address field is displayed correctly
- [ ] Type an invalid IP address into that field and press tab
- [ ] Ensure an error is displayed
- [ ] Go to machine list and click "Add machine"
- [ ] For power type, select [IPMI/LXD/Virsh]
- [ ] Ensure that the IP address field is displayed correctly
- [ ] Ensure that invalid IP addresses cause an error to be shown in this field

### Before
image
image

### After
image
image

## Notes

The MP for the backend implementation of this is [here](https://code.launchpad.net/~alemar99/maas/+git/maas/+merge/466228)

### CODE REVIEW

**Comment by ndv99** [comment]
Hm, I'm getting some real strange behavior in tests surrounding a couple of selectors

`src/app/subnets/views/VLANDetails/VLANDetails.tsx:39:19`
```ts
const subnets = useSelector((state: RootState) =>
    subnetSelectors.getByIds(state, vlan?.subnet_ids || [])
  );
```

`src/app/machines/views/MachineDetails/MachineConfiguration/TagForm/TagForm.tsx:28:18`
```ts
 const errors = useSelector((state: RootState) =>
    machineSelectors.eventErrorsForIds(state, systemId, [
      NodeActions.TAG,
      NodeActions.UNTAG,
    ])
  )[0]?.error;
```

Not quite sure what's causing these (or especially how it's related to this PR's changes), @petermakowski could you take a look if you have a minute?

**Review by petermakowski — COMMENTED** [review]
Have you checked the bundle size impact of adding a new dependency?

**Comment by ndv99** [comment]
> Have you checked the bundle size impact of adding a new dependency?

Good point - I'll run a before/after check. Although likelihood is we'll need this anyway for upcoming reserved IP work

**Comment by amylily1011** [comment]
Screenshot 2024-08-30 at 22 04 55

It looks like the IP address field didn't show up for me.

It looks like I'm in the correct branch 
```
From https://github.com/canonical/maas-ui
 * [new ref]             refs/pull/5447/head -> feat-power-field-ip-validation
Switched to branch 'feat-power-field-ip-validation'
```

**Comment by ndv99** [comment]
> Screenshot 2024-08-30 at 22 04 55
> 
> It looks like the IP address field didn't show up for me.
> 
> It looks like I'm in the correct branch
> 
> ```
> From https://github.com/canonical/maas-ui
>  * [new ref]             refs/pull/5447/head -> feat-power-field-ip-validation
> Switched to branch 'feat-power-field-ip-validation'
> ```

Ahh - this is intended behaviour, my QA steps are wrong :) updated this to reflect expected functionality

File: src/app/store/general/utils/powerTypes.ts [hunk 1]
```diff
@@ -50,6 +51,23 @@ export const formatPowerParameters = (
     return params;
   }, {}) || {};
 
+const getPowerFieldSchema = (fieldType: PowerFieldType) => {
```
Comment (tmerten) [inline src/app/store/general/utils/powerTypes.ts:54]: Nice!

**Review by tmerten — APPROVED** [review]
lgtm, QA done.

**Comment by troyanov** [comment]
@ndv99 @tmerten we had to revert this logic from MAAS, because it was breaking things
https://github.com/canonical/maas/pull/343

Maybe you want to cleanup UI logic as well.

**Comment by ndv99** [comment]
> @ndv99 @tmerten we had to revert this logic from MAAS, because it was breaking things [canonical/maas#343](https://github.com/canonical/maas/pull/343)
> 
> Maybe you want to cleanup UI logic as well.

So should we just use a plain text input here instead and not validate the addresses?

**Pull request merged**